### PR TITLE
Refactor to reduce duplication of file naming script text in user settings

### PR DESCRIPTION
**Pull request by rdswift**
# Summary

* This is a…
    * [ ] Bug fix
    * [ ] Feature addition
    * [x] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [x] Other
* **Describe this change in 1-2 sentences**:  Refactor to remove the duplicate "file_naming_format" setting information, and provide a helper function to retrieve the file naming script text based on the user's "selected_file_naming_script_id" setting.

# Problem

* JIRA ticket (_optional_): PICARD-XXX

Information is (potentially) duplicated in the user configuration settings, specifically the currently selected file naming script which shows up in the "file_naming_format" settings as well as one of the elements of the "file_naming_scripts" list.

# Solution

Remove the "file_naming_format" setting and provide a helper function to look up the file naming script from the list of scripts stored in the user settings, based on the "selected_file_naming_script_id" setting.  To improve the lookup efficiency, the "file_naming_scripts" list was converted to a dictionary and renamed "file_renaming_scripts".  Script items stored in the dictionary are stored as dictionaries of the settings of the `FileNamingScript` object rather than as a YAML dump of the settings.

This PR also includes: some cleanup and optimization of the information synchronization between the "File Naming" and "File Naming Script Editor" pages; moving the "File Naming Script Editor" page help URL to `picard.constants` with the other URL definitions; and uses the `DEFAULT_FILE_NAMING_FORMAT` constant for new file naming scripts rather than "`$noop()`".

Changes include:
- Remove "file_naming_format" setting.
- Add helper function to get the text of the currently selected file naming script.
- Change "file_naming_scripts" list of scripts in YAML format to "file_renaming_scripts" dictionary of `FileNamingScript().to_dict()` format items.
- Use script dictionary objects internally (e.g. in combo box items) rather than full `FileNamingScript` objects.
- Present script combo box lists in sorted order by title.
- Refactor synchronization between "File Naming" and "Script Editor" pages to reduce redundant (multiple) processing.
- Move "Script Editor" page help URL definition to `picard.constants` module with other URL definitions.
- Use `DEFAULT_FILE_NAMING_FORMAT` for new file naming scripts rather than `$noop()`.
- Update function docstrings

# Action

Sorry for the large PR, but most of the changes are interrelated and I struggled trying to find a way to separate them into logical, stand-alone chunks.

### CODE REVIEW

**Review by phw — APPROVED** [review]
Good move

File: picard/script/__init__.py [hunk 1]
```diff
@@ -105,6 +106,27 @@ def enabled_tagger_scripts_texts():
...
+        str: The text of the file naming script if available, otherwise None
+    """
+    from picard.script import get_file_naming_script_presets
+    if settings["file_renaming_scripts"] and settings["selected_file_naming_script_id"]:
+        selected_id = settings["selected_file_naming_script_id"]
+        if selected_id in settings["file_renaming_scripts"]:
+            return settings["file_renaming_scripts"][selected_id]["script"]
+        for item in get_file_naming_script_presets():
+            if item["id"] == selected_id:
+                return str(item["script"])
```
Comment (zas) [inline picard/script/__init__.py:125]: This part would benefit using intermediate variables:

```python
    scripts = settings["file_renaming_scripts"]
    selected_id = settings["selected_file_naming_script_id"]
    if scripts and selected_id:
        if selected_id in scripts:
            return scripts[selected_id]["script"]
        for item in get_file_naming_script_presets():
            if item["id"] == selected_id:
                return str(item["script"])
```

File: picard/script/serializer.py [hunk 2]
```diff
@@ -166,6 +166,38 @@ def update_from_dict(self, settings):
...
             self.update_last_updated()
 
+    def to_dict(self):
+        """Generate a dictionary containing the object's OUTPUT_FIELDS.
+
+        Returns:
+            dict: Dictionary of the object's OUTPUT_FIELDS
+        """
+        items = {key: getattr(self, key) for key in self.OUTPUT_FIELDS}
+        items["script"] = str(items["script"])
```
Comment (zas) [inline picard/script/serializer.py:176]: Why is this `str()` needed? Aren't they always strings?

File: picard/script/serializer.py [hunk 3]
```diff
@@ -166,6 +166,38 @@ def update_from_dict(self, settings):
...
+
+        Args:
+            script_dict (dict): Dictionary containing the property settings.
+            create_new_id (bool, optional): Determines whether a new ID is generated. Defaults to True.
+
+        Returns:
+            object: An instance of the class, populated from the property settings in the dictionary provided.
+        """
+        new_object = cls()
+        if not isinstance(script_dict, dict):
```
Comment (zas) [inline picard/script/serializer.py:192]: `isinstance(script_dict, Mapping)` is more pythonic (any object acting as dict can be used).
https://www.python.org/dev/peps/pep-3119/#mappings

File: picard/ui/options/renaming.py [hunk 4]
```diff
@@ -152,24 +150,29 @@ def update_selector_from_editor(self):
...
         if self.script_editor_dialog:
-            self.script_editor_dialog.set_selected_script_index(self.ui.naming_script_selector.currentIndex(), send_signal=False)
-        self.script_text = script_item["script"]
-        self.selected_naming_script_id = script_item["id"]
-        override = {'file_naming_format': self.script_text}
-        self.examples.update_examples(override)
-        self.update_examples_from_local()
+            self.script_editor_dialog.set_selected_script_index(self.ui.naming_script_selector.currentIndex(), skip_check=skip_check)
+        else:
+            script_item = self.ui.naming_script_selector.itemData(self.ui.naming_script_selector.currentIndex())
```
Comment (zas) [inline picard/ui/options/renaming.py:171]: Store `self.ui.naming_script_selector.currentIndex()` in a variable and shorten lines above

**Review by zas — CHANGES_REQUESTED** [review]
Overall a very good move, just few stylistic comments

File: picard/script/serializer.py (hunk 2 above)
Comment (rdswift) [reply picard/script/serializer.py:176]: Actually the "script" value is returned as a `MultilineLiteral`.

File: picard/script/serializer.py (hunk 3 above)
Comment (rdswift) [reply picard/script/serializer.py:192]: Now I remember you mentioning that to me before.  I had forgotten.  Sorry about that.  I'll make the change.

File: picard/script/serializer.py (hunk 2 above)
Comment (zas) [reply picard/script/serializer.py:177]: Ah ok :)

File: picard/script/__init__.py [hunk 5]
```diff
@@ -116,10 +116,11 @@ def get_file_naming_script(settings):
         str: The text of the file naming script if available, otherwise None
     """
     from picard.script import get_file_naming_script_presets
-    if settings["file_renaming_scripts"] and settings["selected_file_naming_script_id"]:
-        selected_id = settings["selected_file_naming_script_id"]
-        if selected_id in settings["file_renaming_scripts"]:
-            return settings["file_renaming_scripts"][selected_id]["script"]
+    scripts = settings["file_renaming_scripts"]
+    selected_id = settings["selected_file_naming_script_id"]
```
Comment (zas) [inline picard/script/__init__.py:120]: You can also replace the occurence in ` log.error()` below.

**Review by zas — CHANGES_REQUESTED** [review]

File: picard/script/__init__.py (hunk 5 above)
Comment (rdswift) [reply picard/script/__init__.py:120]: Argh!  I completely missed that.  My only excuse is that I just woke up (but maybe not quite it seems).  :blush:

**Review by zas — APPROVED** [review]
LGTM

**Review by phw — APPROVED** [review]

**Pull request merged**